pkg/process/metadata/workloadmeta/workloadmeta.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ProcessEntity is a placeholder workloadmeta.ProcessEntity.
+// It does not contain all the fields that the final entity will contain.
+type ProcessEntity struct {
+	pid      int32
+	language *languagemodels.Language
+}
+
+// WorkloadMetaExtractor handles enriching processes with languages as well as sending enriched processes to the core agent
+// via grpc stream.
+type WorkloadMetaExtractor struct {
+	// Cache is a map from process hash to the workloadmeta entity
+	// The cache key takes the form of `pid:<pid>|createTime:<createTime>`. See hashProcess
+	cache      map[string]*ProcessEntity
+	cacheMutex sync.RWMutex
+
+	grpcListener mockableGrpcListener
+}
+
+// NewWorkloadMetaExtractor constructs the WorkloadMetaExtractor.
+func NewWorkloadMetaExtractor() *WorkloadMetaExtractor {
+	log.Debug("Instantiated the WorkloadMetaExtractor")
+	return &WorkloadMetaExtractor{
+		cache:        make(map[string]*ProcessEntity),
+		grpcListener: newGrpcListener(),
+	}
+}
+
+// Extract detects the process language, creates a process entity, and sends that entity to WorkloadMeta
```

Comment:
We should probably update this comment here since Extract:
- detects the language of processes that have been created and send them to workloadmeta
- detect processes that terminated and send their ID to workloadmeta

Reply:
I agree, that is a good description. I updated the docstring